File: Easy/string_pairs.py
# https://edabit.com/challenge/ZDSrKXGH4dG9M4iuP

def string_pairs(s):
    c= [s[i]+s[i+1] for i in range(0,len(s)-1,2)]
    if len(s)%2==1:
        c.append(s[len(s)-1]+"*")
    return c

def string_pairs(s):
    return [s[i]+s[i+1] if len(s)%2==0 else s[-1]+"*" for i in range(0,len(s)-1,2)]

print(string_pairs("abcdef"), ["ab", "cd", "ef"])
print(string_pairs("abcdefg"), ["ab", "cd", "ef", "g*"])

File: javascript/Very_Easy/stringPairs.js
// https://edabit.com/challenge/tSBCQQ8w9dAAZuKio

function stringPairs(str) {
    const list = []
    for (let index = 0; index <= str.length; index+=2) {
        if(str.slice(index,index+2))list.push(str.slice(index,index+2))
        
    }
    if (str.length%2===1){list[list.length-1]=list[list.length-1]+'*'}
    return list
}

console.log(stringPairs("abcdef"), ["ab", "cd", "ef"])
console.log(stringPairs("abcdefg"), ["ab", "cd", "ef", "g*"])
console.log(stringPairs(""), [])
console.log(stringPairs("pak"), ["pa", "k*"])